Source/MLXNN/Module.swift

```diff
@@ -443,6 +443,10 @@ open class Module {
 
             switch (item, value) {
             case (.value(.parameters(let p)), .value(let newArray)):
+                if p.shape != newArray.shape, verify.contains(.all) {
```

Comment:
I would check `verify.contains(.all)` before comparing the shapes -- if it is disabled we can avoid that work.  it isn't super expensive, but it does require memory allocations:

```
    public var shape: [Int] {
        let ndim = mlx_array_ndim(ctx)
        guard ndim > 0 else { return [] }
        let cShape = mlx_array_shape(ctx)!
        return (0 ..< ndim).map { Int(cShape[$0]) }
    }
```

Reply:
Makes sense https://github.com/ml-explore/mlx-swift/pull/123/commits/f442d76d82b2d63c55469e974fefcf1f46c0f327